sdks/python/apache_beam/io/textio_test.py

```diff
@@ -1442,6 +1461,18 @@ def test_read_escaped_escapechar_after_splitting_many(self):
     source_test_utils.assert_sources_equal_reference_source(
         reference_source_info, sources_info)
 
+  def test_read_from_text_dirname_error(self):
+    file_pattern = 'abcd_test'
+    expected_error_message = (
+        "Path type should be local path or GCS path "
+        "when calling ReadFromText")
+
+    with unittest.mock.patch('os.path.dirname') as mock_dirname:
+      mock_dirname.side_effect = TypeError(expected_error_message)
```

Comment:
Is it possible to actually cause this error, instead? This makes the test possibly masking things if the presumed behavior of `os.path.dirname` is not quite right, or if it changes.